### PR TITLE
Migrate Dataproc to new system tests design

### PR DESCRIPTION
**Pull request by bhirsz**
Migrate Dataproc system tests/example dags to be compliant with new design ([AIP-47](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-47+New+design+of+Airflow+System+Tests)). 

Covers airflow/providers/google/cloud/example_dags/example_dataproc.py from #22447

### CODE REVIEW

File: tests/always/test_project_structure.py [hunk 1]
```diff
@@ -244,10 +243,15 @@ def test_missing_example_for_operator(self):
             }
             all_operators -= example_paths
 
+        covered_but_omitted = self.MISSING_EXAMPLES_FOR_OPERATORS - all_operators
         all_operators -= self.MISSING_EXAMPLES_FOR_OPERATORS
         all_operators -= self.DEPRECATED_OPERATORS
         all_operators -= self.BASE_OPERATORS
-        assert set() == all_operators
+        assert set() == all_operators, (
```
Comment (bhirsz) [inline tests/always/test_project_structure.py]: Also updated the tests to fail if some operators are set as not covered with examples but example is found.

File: tests/system/providers/google/dataproc/example_dataproc_workflow.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    )
+    # [END how_to_cloud_dataproc_trigger_workflow_template]
+
+    # [START how_to_cloud_dataproc_instantiate_inline_workflow_template]
+    instantiate_inline_workflow_template = DataprocInstantiateInlineWorkflowTemplateOperator(
+        task_id='instantiate_inline_workflow_template', template=WORKFLOW_TEMPLATE, region=REGION
+    )
+    # [END how_to_cloud_dataproc_instantiate_inline_workflow_template]
+
+    create_workflow_template >> trigger_workflow
```
Comment (bhirsz) [inline tests/system/providers/google/dataproc/example_dataproc_workflow.py]: I see I missed instantiate_inline_workflow_template from the dependencies, I will update it.

**Review by potiuk — APPROVED** [review]
So much nicer!

**Comment by potiuk** [comment]
Running tests now.

**Comment by potiuk** [comment]
Tests are failing I am afraid

**Comment by bhirsz** [comment]
> Tests are failing I am afraid

The PR stalled over time a bit - I will fix it.

**Review by potiuk — APPROVED** [review]

**Pull request merged**